kernel/comps/virtio/src/device/input/device.rs

```diff
@@ -167,25 +167,45 @@ impl InputDevice {
         }
     }
 
-    /// Query a specific piece of information by `select` and `subsel`, and write
-    /// result to `out`, return the result size.
-    pub fn query_config_select(&self, select: InputConfigSelect, subsel: u8, out: &mut [u8]) -> u8 {
+    pub fn query_config_id_name(&self) -> String {
+        let size = self.select_config(InputConfigSelect::IdName, 0);
+
+        let out = {
+            // TODO: Add a general API to read this byte-by-byte.
+            let mut out = Vec::with_capacity(size);
+            let mut data_ptr = field_ptr!(&self.config, VirtioInputConfig, data).cast::<u8>();
+            for _ in 0..size {
+                out.push(data_ptr.read_once().unwrap());
+                data_ptr.byte_add(1);
+            }
+            out
+        };
+
+        String::from_utf8(out).unwrap()
+    }
+
+    pub fn query_config_prop_bits(&self) -> Option<InputProp> {
+        let size = self.select_config(InputConfigSelect::PropBits, 0);
+        if size == 0 {
+            return None;
+        }
+        assert!(size == 1);
+
+        let data_ptr = field_ptr!(&self.config, VirtioInputConfig, data);
+        InputProp::from_bits(data_ptr.clone().cast::<u8>().read_once().unwrap())
```

Comment:
This `clone` is missed but should be removed as well..

(P.S. If necessary, you can also resolve all my comments above to make the history more readable. GitHub only allows PR authors or privileged maintainers to resolve conversions, which is a stupid design, IMO).